### PR TITLE
Better handling for namespaced components

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Companion PR to https://github.com/withastro/astro/pull/4272
- Updates how we match namespaced `components.with.deep.members`.
- Properly match `*` exports and `import { components } from 'file'`
- Removes `$$metadata` matcher utils which are mostly dead code paths now. With this PR, they can probably be removed from the frontend!

## Testing

Tests updated

## Docs

Bug fix only

### CODE REVIEW

File: internal/transform/transform.go [hunk 1]
```diff
@@ -288,11 +288,6 @@ func ExtractScript(doc *astro.Node, n *astro.Node, opts *TransformOptions) {
 func AddComponentProps(doc *astro.Node, n *astro.Node, opts *TransformOptions) {
 	if n.Type == astro.ElementNode && (n.Component || n.CustomElement) {
 		for _, attr := range n.Attr {
-			id := n.Data
-			if n.CustomElement {
```
Comment (matthewp) [inline internal/transform/transform.go:292]: this doesn't look related to this change, are you sure it's safe to remove this? I know we don't support using CE tag names any more but I think we still print them to go through renderComponent, right?

Comment (natemoo-re) [reply internal/transform/transform.go:292]: Yep, thanks for checking! After the refactoring below, `id` is unused. We still handle CE tags here, but we're also handling Component tags the same way now (as a string)

Comment (matthewp) [reply internal/transform/transform.go:292]: Oh got it, since we are not using metadata.getPath we don't need the id here at all. Thanks.

**Review by matthewp — APPROVED** [review]
